google/cloud/datastore/_api.py

```diff
@@ -348,3 +356,27 @@ def allocate_ids(self, project, request_pb):
         """
         request_pb.project_id = project
         return self._stub.AllocateIds(request_pb)
+
+
+def build_api_url(project, method, base_url):
```

Comment:
Would it make more sense to move this function up near the constants which govern its behavior?